### PR TITLE
ref(replay): Refactor extractDomNodes & countDomNodes to return an indexed map

### PR DESCRIPTION
**Pull request by ryan953**
I want to use an indexed map to more easily merge extractedDomNode information with traces inside one tab.

This refactor changes the return type for `extractDomNodes` to do that. Also `countDomNodes` to keep things consistent. 

I went and made a base class for them both because i noticed that `createHiddenPlayer` was copy+pasted, and then got carried away.

Related to https://github.com/getsentry/sentry/issues/50065

### CODE REVIEW

File: static/app/utils/replays/replayReader.tsx [hunk 1]
```diff
@@ -259,21 +257,6 @@ export default class ReplayReader {
...
 
-  countDomNodes = memoize(() =>
-    countDomNodes({
-      frames: this.getRRWebMutations(),
-      rrwebEvents: this.getRRWebFrames(),
-      startTimestampMs: this._replayRecord.started_at.getTime(),
-    })
-  );
-
-  getDomNodes = memoize(() =>
```
Comment (ryan953) [inline static/app/utils/replays/replayReader.tsx:270]: this was memoized here, but it's also memoized again when we leverage `useApiQuery` to await the promise. 

So i thought, we can remove this layer of the callstack.

File: static/app/utils/replays/replayReader.tsx [hunk 2]
```diff
@@ -259,21 +257,6 @@ export default class ReplayReader {
...
-    extractDomNodes({
-      frames: this.getDOMFrames(),
-      rrwebEvents: this.getRRWebFrames(),
-    })
-  );
```
Comment (ryan953) [inline static/app/utils/replays/replayReader.tsx:275]: These are both memoized here, but also memoized again when we leverage `useApiQuery` to await the promises.

So i thought, we can remove this layer of the callstack.

File: static/app/utils/replays/replayerStepper.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  domRoot.className = 'sentry-block';
+  const {style} = domRoot;
+
+  style.position = 'fixed';
+  style.inset = '0';
+  style.width = '0';
+  style.height = '0';
+  style.overflow = 'hidden';
+
+  document.body.appendChild(domRoot);
```
Comment (billyvg) [inline static/app/utils/replays/replayerStepper.tsx:98]: Does this need to be appended in dom to work?

**Review by billyvg — APPROVED** [review]

File: static/app/utils/replays/replayerStepper.tsx (hunk 3 above)
Comment (ryan953) [reply static/app/utils/replays/replayerStepper.tsx:98]: Ya. something about the replayer needs it, but i'm not sure what.

Comment (ryan953) [reply static/app/utils/replays/replayerStepper.tsx:98]: When it's removed the DOM tab loads quicker, without any html appearing.

**Pull request merged**